src/Nethermind/Nethermind.Blockchain/Producers/BaseBlockProducer.cs

```diff
@@ -59,6 +61,7 @@ public abstract class BaseBlockProducer : IBlockProducer
             BlockTree = blockTree ?? throw new ArgumentNullException(nameof(blockTree));
             BlockProcessingQueue = blockProcessingQueue ?? throw new ArgumentNullException(nameof(blockProcessingQueue));
             _stateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
+            _signer = signer ?? throw new ArgumentNullException(nameof(signer));
```

Comment:
Not all the block producers require signers -> for example Ethash based block prodcution (any PoW will not require any signatures). Hence the signer should be an internal thing of the Sealer